turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -1044,7 +1044,12 @@ func typedSegments(dir string, minBlock uint64, types []snaptype.Type) (res []sn
 				}
 				l = append(l, f)
 			}
-			l, m = noGaps(noOverlaps(segmentsTypeCheck(dir, l)), minBlock)
+
+			if allowMissing {
```

Comment:
why do we need to allow missing?

Reply:
im not sure if we should

Reply:
It allows us to start off from any block number instead of having to always download from the first snapshot to where we want to be.

Reply:
ok I see - could you please then rename it to `allowGaps` instead of `allowMissing` - better matches the exclusion of the `noGaps` function